### ISSUE TITLE
x:Bind OneWay to attached property is evaluated only once and not binding to a private property of a page

### ISSUE THREAD

**Issue by FrozDark** [unoplatform/uno#11606]
### Current behavior

If you use x:Bind to bind to attached property with Mode=OneWay it behaves like it's OneTime and evaluated only once even if property of a target object is changed

According to debug of the Converter it's called only 1 time. Tried to RegisterCallbackChange to determine if DependecyObject work well and it's works so the issue is related direct to x:Bind and is like using OneTime mode

Another issue is if you use private property it unable to retrieve the value of that property (works on WinUI)

![agaga](https://user-images.githubusercontent.com/6670150/223639433-68eead02-8364-4117-848d-d1b1fb6c1ca9.jpg)

### Expected behavior

Work as WinUI

### How to reproduce it (as minimally and precisely as possible)

Repro
[xBindUnoIssues.zip](https://github.com/unoplatform/uno/files/10917926/xBindUnoIssues.zip)

### Workaround

Out of knowledge

### Works on UWP/WinUI

Yes

### Environment

Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### Affected platforms

WebAssembly, Android, iOS, macOS (AppKit), Mac Catalyst, Skia (WPF), Skia (GTK on Linux/macOS/Windows), Skia (Linux Framebuffer), Skia (Tizen)

### IDE

Visual Studio 2022

**Comment by MartinZikmund** [comment]
Still relevant

**Comment by MartinZikmund** [comment]
# Repro generator

Runtime test(s) added in branch [`dev/mazi/repro/11606`](https://github.com/unoplatform/uno/compare/dev/mazi/repro/11606):
- `src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml/Given_xBind.cs` → `When_xBind_OneWay_AttachedProperty_Updates`

Test fails with `Assert.AreEqual failed` — after changing an attached property value from "InitialValue" to "UpdatedValue", the x:Bind OneWay binding still reads "InitialValue", confirming the binding behaves as OneTime instead of OneWay for attached properties.
